tests/test_setup.py

```diff
@@ -27,6 +29,12 @@ def setUp(self) -> None:
         # Force the most restrictive interpretation.
         self.set_environ('LC_ALL', 'C')
 
+    def raise_error(self, error: type[Exception]) -> Callable[[Any], Any]:
```

Comment:
This seems like something to use unittest.mock or pytest's monkeypatch for.  But see below.